smsclient_core/smsclient.py

```diff
@@ -0,0 +1,401 @@
+
+    name = fields.Text('SMS Request', size=256,
+                       required=True, readonly=True,
+                       states={'draft': [('readonly', False)]})
+    msg = fields.Text('SMS Text', size=256,
+                      required=True, readonly=True,
+                      states={'draft': [('readonly', False)]})
+    mobile = fields.Char('Mobile No', size=256,
+                         required=True, readonly=True,
+                         states={'draft': [('readonly', False)]})
+    gateway_id = fields.Many2one('sms.smsclient',
+                                 'SMS Gateway', readonly=True,
+                                 states={'draft': [('readonly', False)]})
+    state = fields.Selection([
+        ('draft', 'Queued'),
+        ('sending', 'Waiting'),
+        ('send', 'Sent'),
+        ('error', 'Error'),
+    ], 'Message Status', select=True, readonly=True, default='draft')
+    error = fields.Text('Last Error', size=256,
+                        readonly=True,
+                        states={'draft': [('readonly', False)]})
+    date_create = fields.Datetime('Date', readonly=True,
+                                  default=fields.Datetime.now)
+    validity = fields.Integer(
+        'Validity',
+        help='The maximum time -in minute(s)- before the message is dropped')
+
+    classes = fields.Selection(
+        CLASSES_LIST,
+        'Class',
+        help='The sms class: flash(0), phone display(1), SIM(2), toolkit(3)')
+    deferred = fields.Integer(
+        'Deferred',
+        help='The time -in minute(s)- to wait before sending the message')
+
+    priority = fields.Selection(PRIORITY_LIST, 'Priority',
+                                help='The priority of the message ')
+    coding = fields.Selection([
+        ('1', '7 bit'),
+        ('2', 'Unicode')
+        ], 'Coding', help='The sms coding: 1 for 7 bit or 2 for unicode')
+    tag = fields.Char('Tag', size=256,
+                      help='An optional tag')
+    nostop = fields.Boolean(
+        'NoStop',
+        help='Do not display STOP clause in the message, this requires that'
+             'this is not an advertising message')
+
+
+class Properties(models.Model):
+    _name = 'sms.smsclient.parms'
+    _description = 'SMS Client Properties'
+
+    name = fields.Char('Property name', size=256,
+                       help='Name of the property whom appear on the URL')
+    value = fields.Char('Property value', size=256,
+                        help='Value associate on the property for the URL')
+    gateway_id = fields.Many2one('sms.smsclient', 'SMS Gateway')
+    type = fields.Selection([
+        ('user', 'User'),
+        ('password', 'Password'),
+        ('sender', 'Sender Name'),
+        ('to', 'Recipient No'),
+        ('sms', 'SMS Message'),
+        ('extra', 'Extra Info')
+        ], 'API Method', select=True,
+        help='If parameter concern a value to substitute, indicate it')
+
+
+class HistoryLine(models.Model):
```

Comment:
No sure that an HistoryLine object is needed. We can just keep the mail send in the queue with the state send for the history, what do you think?